### PR TITLE
Store player names in teehistorian and Finish/TeamFinish

### PR DESCRIPTION
**Pull request by Zwelf**
## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [x] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [x] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by Zwelf** [comment]
For me it would be best if #8220 #8221 and #8171 simultaneously so that the version bump in teehistorian_minor would cover all three PRs.

**Review by def- — APPROVED** [review]

File: src/game/server/score.cpp [hunk 1]
```diff
@@ -160,15 +162,15 @@ void CScore::SaveScore(int ClientId, float Time, const char *pTimestamp, const f
 	FormatUuid(GameServer()->GameUuid(), Tmp->m_aGameUuid, sizeof(Tmp->m_aGameUuid));
 	Tmp->m_ClientId = ClientId;
 	str_copy(Tmp->m_aName, Server()->ClientName(ClientId), sizeof(Tmp->m_aName));
-	Tmp->m_Time = Time;
+	Tmp->m_Time = (float)(TimeTicks) / (float)Server()->TickSpeed();
```
[the remark this answers is not in the document]
Comment (Zwelf) [reply src/game/server/score.cpp:165]: Oh, sorry about that. Thanks for the fast revert.

File: src/test/teehistorian.cpp [hunk 2]
```diff
@@ -838,6 +838,53 @@ TEST_F(TeeHistorian, AntibotEmptyMessage)
 	Expect(EXPECTED, sizeof(EXPECTED));
 }
 
+TEST_F(TeeHistorian, PlayerName)
+{
+	const unsigned char EXPECTED[] = {
+		// EX uuid=d016f9b9-4151-3b87-87e5-3a6087eb5f26 datalen=4
```
Comment (heinrich5991) [inline src/test/teehistorian.cpp:844]: Here and below, `datalen=4` seems to be incorrect.

File: src/game/server/teehistorian.cpp [hunk 3]
```diff
@@ -573,6 +573,23 @@ void CTeeHistorian::RecordPlayerDrop(int ClientId, const char *pReason)
...
+	EnsureTickWritten();
+
+	CPacker Buffer;
+	Buffer.Reset();
+	Buffer.AddInt(ClientId);
+	Buffer.AddString(pName, 0);
+
+	if(m_Debug)
+	{
+		dbg_msg("teehistorian", "drop cid=%d reason='%s'", ClientId, pName);
```
Comment (heinrich5991) [inline src/game/server/teehistorian.cpp:587]: This seems to be wrong as well.

File: src/game/server/teehistorian.cpp [hunk 4]
```diff
@@ -794,6 +811,34 @@ void CTeeHistorian::RecordAntibot(const void *pData, int DataSize)
...
 
+void CTeeHistorian::RecordPlayerFinish(int ClientId, int TimeTicks)
+{
+	CPacker Buffer;
+	Buffer.Reset();
+	Buffer.AddInt(ClientId);
+	Buffer.AddInt(TimeTicks);
+	if(m_Debug)
+	{
+		dbg_msg("teehistorian", "player_finish cid=%d Time=%d", ClientId, TimeTicks);
```
Comment (heinrich5991) [inline src/game/server/teehistorian.cpp:822]: time lowercase in debug message.

File: src/game/server/teehistorian.cpp [hunk 5]
```diff
@@ -794,6 +811,34 @@ void CTeeHistorian::RecordAntibot(const void *pData, int DataSize)
...
+
+void CTeeHistorian::RecordTeamFinish(int TeamId, int TimeTicks)
+{
+	CPacker Buffer;
+	Buffer.Reset();
+	Buffer.AddInt(TeamId);
+	Buffer.AddInt(TimeTicks);
+	if(m_Debug)
+	{
+		dbg_msg("teehistorian", "player_finish cid=%d Time=%d", TeamId, TimeTicks);
```
Comment (heinrich5991) [inline src/game/server/teehistorian.cpp:836]: `team_finish`

**Pull request merged**